src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -684,19 +684,16 @@ internal sealed override bool IsInterface
             }
         }
 
-        internal override ManagedKind ManagedKind
+        internal override ManagedKind GetManagedKind(ref HashSet<DiagnosticInfo> useSiteDiagnostics)
         {
-            get
+            var managedKind = _flags.ManagedKind;
+            if (managedKind == ManagedKind.Unknown)
             {
-                var managedKind = _flags.ManagedKind;
-                if (managedKind == ManagedKind.Unknown)
-                {
-                    var baseKind = base.ManagedKind;
-                    _flags.SetManagedKind(baseKind);
-                    return baseKind;
-                }
-                return managedKind;
+                var baseKind = base.GetManagedKind(ref useSiteDiagnostics);
+                _flags.SetManagedKind(baseKind);
+                return baseKind;
             }
+            return managedKind;
```

Comment:
>managedKind [](start = 19, length = 11)

It looks like we are dropping use-site diagnostics on the second call. Please add a targeted test for a situation like that. Now we need to cache the value and use-site diagnostics together, or not cache either.

Reply:
> Please add a targeted test for a situation like that

I didn't see a test added. The test I have in mind is an explicit use of the API on SourceMemberContainerSymbol twice and an assert that we get the same set of diagnostic info both times.

---
In reply to: [458396646](https://github.com/dotnet/roslyn/pull/46114#discussion_r458396646) [](ancestors = 458396646)

Reply:
I tested this indirectly by using the type in a way that requires reading IsUnmanaged on the symbol multiple times. Can test the API directly instead.

Reply:
> Can test the API directly instead.

This will ensure that the test doesn't depend on implementation elsewhere in the compiler.

---
In reply to: [458947858](https://github.com/dotnet/roslyn/pull/46114#discussion_r458947858) [](ancestors = 458947858)